File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/ChamberDefinition.java
package vip.qoriginal.quantumplugin.chambers;

import org.bukkit.structure.Structure;

public record ChamberDefinition(
        String id,
        String title,
        String objective,
        Structure structure,
        boolean includeEntities,
        ChamberPosition spawn,
        ChamberRegion goal,
        int timeLimitSeconds,
        ChamberScripts scripts
) {
}

File: chambers/src/main/java/vip/qoriginal/quantumplugin/chambers/data/ChamberDefinition.kt
package vip.qoriginal.quantumplugin.chambers.data

import org.bukkit.structure.Structure
import vip.qoriginal.quantumplugin.chambers.ChamberScripts

data class ChamberDefinition(
	val id: String,
	val title: String,
	val objective: String,
	val structure: Structure,
	val includeEntities: Boolean,
	val spawn: ChamberPosition,
	val goal: ChamberRegion,
	val timeLimitSeconds: Int,
	val scripts: ChamberScripts,
)